### ISSUE TITLE
cocos2d-x 3.11.1 prebuild failed using clang & NDK r12b

### ISSUE THREAD

**Issue by scp-r** [cocos2d/cocos2d-x#16033]
engine ver: 3.11.1
ndk : r12b
OS : Windows 10 x64
android api level :15

in order to use clang, i edited build_android.py according to this:
https://github.com/cocos2d/cocos2d-console/pull/350

and edited tools\simulator\libsimulator\proj.android\Android.mk according to http://blog.csdn.net/final5788/article/details/51433590

```
include $(CLEAR_VARS)

LOCAL_MODULE    := cocos2d_lua_static

LOCAL_MODULE_FILENAME := libluacocos2d
#modified begin
ifeq ($(COCOS_SIMULATOR_BUILD),1)
LOCAL_ARM_MODE := arm
endif
#modified end
LOCAL_SRC_FILES := ../manual/CCLuaBridge.cpp \
```

and run:

```
cocos gen-libs -p android
```

error msg:

```
[armeabi] StaticLibrary  : libnetwork.a
[armeabi] StaticLibrary  : libcocos2dxinternal.a
[armeabi] StaticLibrary  : librecast.a
[armeabi] StaticLibrary  : libbullet.a
[armeabi] StaticLibrary  : libluacocos2dandroid.a
[armeabi] StaticLibrary  : libjscocos2dandroid.a
[armeabi] StaticLibrary  : libcpufeatures.a
[armeabi] StaticLibrary  : libcocos2dandroid.a
[armeabi] SharedLibrary  : libcocos2dlua.so
G:/android-ndk-r12b/build//../sources/cxx-stl/gnu-libstdc++/4.9/include\bits/hashtable.h:802: error: relocation overflow
 in R_ARM_THM_CALL
G:/android-ndk-r12b/build//../sources/cxx-stl/gnu-libstdc++/4.9/include\bits/hashtable.h:802: error: relocation overflow
 in R_ARM_THM_CALL
G:/android-ndk-r12b/build//../sources/cxx-stl/gnu-libstdc++/4.9/include\bits/hashtable.h:802: error: relocation overflow
 in R_ARM_THM_CALL
G:/android-ndk-r12b/build//../sources/cxx-stl/gnu-libstdc++/4.9/include\bits/hashtable.h:802: error: relocation overflow
 in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/ext/concurrence.h:219: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/ext/atomicity.h:49: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/bits/basic_string.h:249: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/ext/atomicity.h:49: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/bits/basic_string.h:249: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/ext/atomicity.h:49: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/bits/basic_string.h:249: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/ext/atomicity.h:49: error: relocation overflow in R_ARM_THM_CALL
/usr/local/google/buildbot/src/android/ndk-r12-release/out/build/tmp/build-16229/build-gnustl/static-armeabij4.9/include
/bits/basic_string.h:249: error: relocation overflow in R_ARM_THM_CALL
clang++.exe: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [obj/local/armeabi/libcocos2dlua.so] Error 1
make: Leaving directory `C:/Software/cocos2d-x-3.11.1/tools/simulator/frameworks/runtime-src/proj.android'
执行命令出错，返回值：2。
执行命令出错，返回值：14。
```

how to fix it?Thank you for any help.

**Comment by scp-r** [comment]
plus, if use gcc,  everything works.

**Comment by minggo** [comment]
I will download NDK r12b to take a look. I suggest you using older NDK, such as r10c.

**Comment by minggo** [comment]
I mean i suggest you using older NDK version to work around it currently.

**Comment by minggo** [comment]
Yep, i met the issue with NDK r12b. I will try to fix it.

**Comment by minggo** [comment]
I asked help from google guys, and they replied

```
Based on my communication with NDK engineers, it seems that the linker is trying to relocate against an address that is too far away to be encoded into the relocation.
Typically happens in a huge library. Similar problem you can see in this with different wording for different linkers.
There is also a good description here: https://groups.google.com/forum/#!topic/android-ndk/8w112eP9W3k
And the workaround using ARM mode in Android.mk seems to be just delaying the problem.

The best way proposed from the NDK team to resolve this issue was that you file a bug directly on https://github.com/android-ndk/ndk/issues.
```

So it is a bug of NDK. I will create an issue in NDK github issue system and put a link here. Of course, i will try to delay the error happen.

**Comment by minggo** [comment]
I created an issue in NDK repo [here](https://github.com/android-ndk/ndk/issues/152).

**Comment by minggo** [comment]
fixed in #16069, as google guys said it just delay the issue, will update if NDK engineers fix the issue
